### PR TITLE
feat: update to use OpenTelemetry.Tracer

### PR DESCRIPTION
**Pull request by forest**
Also use SemanticConventions where possible.

### CODE REVIEW

**Comment by forest** [comment]
`{instrumentation_scope,<<"opentelemetry_avalanche">>,<<"0.0.0">>}` Is that we are looking for.

```
*SPANS FOR DEBUG*
{span,216687236971084032639096624498985333956,8857844433121593954,[],
      17937131057303637203,<<"avalanche.query">>,client,-576460736185074918,
      -576460717446667176,
      {attributes,128,infinity,0,
                  #{'db.error' => nil,'db.num_rows' => nil,'db.result' => nil,
                    'db.statement' =>
                        <<"...">>,
                    'db.system' => snowflake,total_time_microseconds => 0}},
      {events,128,128,infinity,0,[]},
      {links,128,128,infinity,0,[]},
      undefined,1,false,
      {instrumentation_scope,<<"opentelemetry_avalanche">>,<<"0.0.0">>,
                             undefined}}
```

File: lib/opentelemetry_avalanche.ex [hunk 1]
```diff
@@ -6,35 +6,47 @@ defmodule OpentelemetryAvalanche do
...
+                      type: :string,
+                      default: "avalanche.query",
+                      doc: "The name of the span."
+                    ],
+                    trace_query_statement: [
+                      type: :boolean,
+                      default: true,
+                      doc: "Should trace query statement?"
+                    ],
+                    trace_query_params: [
```
Comment (pdgonzalez872) [inline lib/opentelemetry_avalanche.ex:28]: maybe we add a comentadjust the doc_string below saying: "be careful with allowing this as you may have sensitive data here"

File: lib/opentelemetry_avalanche.ex [hunk 2]
```diff
@@ -88,18 +100,14 @@ defmodule OpentelemetryAvalanche do
...
-        {:"db.statement", metadata[:query]}
-      )
-      |> put_if(config.trace_query_params, {:"db.params", inspect(metadata[:params])})
-
-    start_opts = %{kind: :client}
-
-    @tracer_id
-    |> OpentelemetryTelemetry.start_telemetry_span(config.span_name, metadata, start_opts)
-    |> Span.set_attributes(attributes)
+      [{SemanticConventions.Trace.db_system(), :snowflake}]
```
Comment (pdgonzalez872) [inline lib/opentelemetry_avalanche.ex:103]: very interesting

**Review by pdgonzalez872 — APPROVED** [review]

**Pull request merged**